### PR TITLE
editor: confirmation message when leaving

### PR DESCRIPTION
**Pull request by Garfield-fr**
A confirmation message is displayed if the user leaves the form without saving it.

* Removes spinner on editor.
* Changes the visiblity of the property in the editor to allow
  abstracting the class.
* Closes rero/rero-ils#2104.

### CODE REVIEW

**Comment by PascalRepond** [comment]
## PO tests

-  [ ] REP: In all editors (except library editor), the confirmation message displays multiple times (as much as it has been used+1) before successfully leaving the page.

**Review by PascalRepond — APPROVED** [review]
Works perfectly, great job!

**Review by zannkukai — APPROVED** [review]

File: projects/rero/ng-core/src/lib/guard/component-can-deactivate.guard.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   * @param _translateService - TranslateService
+   * @param _dialogService - DialogService
+   */
+  constructor(
+    protected translateService: TranslateService,
+    protected dialogService: DialogService
+  ) {}
+
+  /**
+   * CanDeactivate
```
Comment (jma) [inline projects/rero/ng-core/src/lib/guard/component-can-deactivate.guard.ts:38]: Can you put more details on the description?

File: projects/rero/ng-core/src/lib/record/editor/editor.component.ts [hunk 2]
```diff
@@ -150,33 +156,32 @@ export class EditorComponent implements OnInit, OnChanges, OnDestroy {
...
-    private _route: ActivatedRoute,
-    private _recordUiService: RecordUiService,
-    private _translateService: TranslateService,
-    private _toastrService: ToastrService,
-    private _location: Location,
-    private _spinner: NgxSpinnerService,
-    private _modalService: BsModalService,
-    private _routeCollectionService: RouteCollectionService,
-    private _loggerService: LoggerService
+    protected formlyJsonschema: FormlyJsonschema,
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/editor/editor.component.ts:172]: Can you add an entry in you commit message?

**Review by jma — APPROVED** [review]

**Pull request merged**